ruby/bindings/InitRubyBindings.cpp

```diff
@@ -715,13 +756,19 @@ void locateEmbeddedGems(bool use_bundler) {
 
     embedded_gems_to_activate.each do |spec|
       if spec.name == "bundler"
-        $logger.debug "Activating gem #{spec.spec_file}"
+        $logger.debug "Activating Bundler gem #{spec.spec_file}"
+        if $logger.trace?
+          pp spec
+        end
         begin
           # Activate will manipulate the $LOAD_PATH to include the gem
           spec.activate
-        rescue Gem::LoadError
+        rescue Gem::LoadError => e
           # There may be conflicts between the bundle and the embedded gems,
           # those will be logged here
+          exception_msg = "Error activating gem #{spec.spec_file}: #{e.class}: #{e.message}\nTraceback:\n"
+          exception_msg += e.backtrace.join("\n")
+          STDERR.puts exception_msg
```

Comment:
No matter what I do, I can't get the failure like I do in the ruby cli.

```shell
cd OpenStudio/src/cli/test/bundle
ruby --version
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
bundle install --path ./test_gems
```

**C++**

```shell
$ $os_build_rel/Products/openstudio --bundle Gemfile.lock --bundle_path './test_gems' --loglevel Debug test.rb
[...]
[ruby] <-2> Adding dependency on tbd '~> 3.4.1'
[ruby] <-2> Adding dependency on topolys '~> 0.6.2'
[ruby] <-2> Adding dependency on unicode-display_width '~> 2.5.0'
[ruby] <-2> Adding dependency on uuid '~> 2.3.9'
[ruby] <-2> Adding dependency on yamler '~> 0.1.0'
[ruby] <-2> Adding dependency on zliby '~> 0.0.5'
[ruby] <-2> Processing 136 activation requests
[ruby] <-2> Activating Bundler gem :/ruby/3.2.0/specifications/bundler-2.5.5.gemspec
Failed to execute '/home/julien/Software/Others/OpenStudio/src/cli/test/bundle/test.rb'
```

**Classic**

```shell
$ $os_build_rel/Products/openstudio classic --bundle Gemfile.lock --bundle_path './test_gems' --verbose test.rb

[...]
D, [2024-04-19T14:52:29.102995 #1611746] DEBUG -- : Processing 136 activation requests
D, [2024-04-19T14:52:29.119140 #1611746] DEBUG -- : Activating gem :/ruby/3.2.0/specifications/bundler-2.5.5.gemspec
Error executing argv: ["--bundle", "Gemfile.lock", "--bundle_path", "./test_gems", "--verbose", "test.rb"]
Error: 
[!] There was an error parsing `Gemfile.lock`: syntax error, unexpected ':', expecting end-of-input -   remote: http://rubygems.org/
        ^
. Bundler cannot continue.

 #  from /home/julien/Software/Others/OpenStudio/src/cli/test/bundle/Gemfile.lock:2
 #  -------------------------------------------
 #  GEM
 >    remote: http://rubygems.org/
 #    specs:
 #  -------------------------------------------
 in :/ruby/3.2.0/gems/bundler-2.5.5/lib/bundler/dsl.rb:49:in `instance_eval'
:/ruby/3.2.0/gems/bundler-2.5.5/lib/bundler/dsl.rb:49:in `eval_gemfile'
:/ruby/3.2.0/gems/bundler-2.5.5/lib/bundler/dsl.rb:12:in `evaluate'
:/ruby/3.2.0/gems/bundler-2.5.5/lib/bundler/definition.rb:38:in `build'
:/ruby/3.2.0/gems/bundler-2.5.5/lib/bundler.rb:326:in `definition'
:/openstudio_cli.rb:645:in `parse_main_args'
:/openstudio_cli.rb:791:in `execute'
:/openstudio_cli.rb:1994:in `<main>'
eval:233:in `eval'
eval:233:in `require_embedded_absolute'
eval:218:in `block in require_embedded'
eval:212:in `each'
eval:212:in `require_embedded'
eval:171:in `require'
eval:3:in `<main>'
```

Reply:
@kbenne that error makes zero sense to me.